1-Draft/RFCNNNN-OutVariableTransparency.md

```diff
@@ -0,0 +1,41 @@
+---
+RFC:
+Author: Robert Holt
+Status: Draft
+SupercededBy: N/A
+Version:
+Area: Language/Common Variables
+Comments Due: 2018-05-01
+Plan to implement: Yes
+---
+
+# OutVariable Transparency
+
+PowerShell supports OutVariables, which allow users to send the output of a command to the variable rather than to the pipeline.
+
+Since this feature was included in PowerShell v1, the implementation detail of the ArrayList used to collect the output into the OutVariable has been exposed to the user, so that rather than getting the `object` or `object[]` that they would get with pipeline output, an `ArrayList` is always received.
+
+This RFC floats the idea that OutVariable usage should be transparent to the user, so output is agnostic between OutVariable and the pipeline stream. As a corner case in PowerShell's implementation, OutVariables currently represent extra caveats for new users.
+
+## Motivation
+
+    As a PowerShell user, I can use OutVariable just like pipeline output, so that my language experience is simpler and more consistent.
+
+## Specification
+
+This table summarises the changes proposed in this RFC:
+
+| Input                                            | Non-OutVar Result Type       | Old Result Type              | New Result Type              |
+| :----------------------------------------------: | :--------------------------: | :--------------------------: | :--------------------------: |
+| `'Hello'`                                        | System.String                | System.Collections.ArrayList | System.String                |
+| `@(1, 2)`                                        | object[]                     | System.Collections.ArrayList | object[]                     |
```

Comment:
Why would we use `Object[]`? we should be moving away from `Object[]` in favor of `List<Object>`. I don't see why taking a step back from `ArrayList` to `Object[]` is beneficial to anyone.